cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -162,8 +162,10 @@ pub struct MachBuffer<I: VCodeInst> {
     /// Latest branches, to facilitate in-place editing for better fallthrough
     /// behavior and empty-block removal.
     latest_branches: SmallVec<[MachBranch; 4]>,
-    /// All labels, in offset order.
-    labels_by_offset: SmallVec<[(MachLabel, CodeOffset); 16]>,
+    /// All labels at the current offset (emission tail). For correctness, this
+    /// *must* be complete, because we rely on it to update labels when we
```

Comment:
What does "complete" mean here?

Reply:
Clarified -- just means that the vector must contain all labels that are bound to this offset (the buffer tail); missing a label would result in that label not being adjusted properly when it needs to be.